src/main/plots/solution_graph_statistics_plots.py

```diff
@@ -0,0 +1,107 @@
+
+
+def plot_general_statistics(solution_graph: SolutionGraph, path: str = DEFAULT_PATH_FOR_PLOTS,
+                            format: consts.EXTENSION = consts.EXTENSION.HTML, auto_open: bool = False,
+                            x_category_order: PLOTTY_CATEGORY_ORDER = PLOTTY_CATEGORY_ORDER.TOTAL_ASCENDING) -> None:
+    canon_trees_statistics, anon_trees_statistics = get_general_solution_graph_statistics(solution_graph)
+    labels = get_labels_for_freq_plots(STATISTICS_KEY.NODES_NUMBER)
+    __plot_general_statistics(canon_trees_statistics, title='Vertices number for canon trees', path=path,
+                              plot_name='vertices_number_for_canon_trees', format=format, auto_open=auto_open,
+                              x_category_order=x_category_order, labels=labels)
+    __plot_general_statistics(anon_trees_statistics, title='Vertices number for anon trees', path=path,
+                              plot_name='vertices_number_for_anon_trees', format=format, auto_open=auto_open,
+                              x_category_order=x_category_order, labels=labels)
+
+
+def __add_canon_nodes_numbers(fig: go.Figure, canon_nodes_number: int,
+                              anon_nodes_number_freq: Dict[int, int]) -> go.Figure:
+    fig.add_trace(
+        go.Scatter(
+            mode='markers',
+            x=[canon_nodes_number],
+            y=[anon_nodes_number_freq.get(canon_nodes_number, 1) / 2],
+            marker=dict(
+                color='White',
+                size=20,
+                line=dict(width=1, color='DarkSlateGrey')
+            ),
+            showlegend=False,
+            marker_symbol='x',
+            hoverinfo='skip'
+        )
+    )
+    return fig
+
+
+def plot_statistics_for_vertex(vertex_id: int, labels: Dict[str, str], anon_nodes_numbers: List[int],
+                               canon_nodes_number: int, path: str, format: consts.EXTENSION, auto_open: bool,
+                               column: STATISTICS_KEY = STATISTICS_KEY.NODES_NUMBER) -> None:
+    title = __get_tittle_for_each_vertex_plots(vertex_id)
+    freq_dict = defaultdict(int)
+    for key, group in groupby(anon_nodes_numbers):
+        freq_dict[key] = len(list(group))
+    statistics_df = pd.DataFrame(freq_dict.items(), columns=[column.value, STATISTICS_FREQ])
+    fig = get_freq_bar_chart(statistics_df, title, column, labels,
+                             x_category_order=PLOTTY_CATEGORY_ORDER.TRACE,
+                             to_update_layout=False)
+    __add_canon_nodes_numbers(fig, canon_nodes_number, freq_dict)
+    # Todo: fix x_category_order, because now order is not always ascending
+    fig = update_layout(fig, column, PLOTTY_CATEGORY_ORDER.TRACE)
+    save_plot(fig, path, PLOT_TYPE.BAR, f'vertex_{vertex_id}', format, auto_open)
+
+
+def __get_tittle_for_each_vertex_plots(vertex_id_: int, column: STATISTICS_KEY = STATISTICS_KEY.NODES_NUMBER) -> str:
```

Comment:
It's a little unclear what do you mean by 'for each vertex'. In the function below you use it when you iterating over all vertices, but in this function, you just get a title for one vertex, which id is passed as an argument. Let's rename it to 'get_title_for_vertex_plot'?